### PR TITLE
Add extra filter for file picker

### PR DESCRIPTION
**Pull request by PVince81**
Makes it possible to be more flexible when filtering entries to be
displayed.

Needed for the import file extension: https://github.com/nextcloud/user_migration/pull/109

### CODE REVIEW

**Review by artonge — APPROVED** [review]

File: core/src/OC/dialogs.js [hunk 1]
```diff
@@ -296,6 +297,9 @@ const Dialogs = {
 				sizeCol: t('core', 'Size'),
 				modifiedCol: t('core', 'Modified')
 			}).data('path', path).data('multiselect', multiselect).data('mimetype', mimetypeFilter).data('allowDirectoryChooser', options.allowDirectoryChooser)
+			if (typeof(options.filter) === 'function') {
+				self.$filePicker.data('filter', options.filter);
```
Comment (CarlSchwan) [inline core/src/OC/dialogs.js:301]:
```suggestion
				self.$filePicker.data('filter', options.filter)
```

File: core/src/OC/dialogs.js [hunk 2]
```diff
@@ -1142,6 +1147,10 @@ const Dialogs = {
 				})
 			}
 
+			if (advancedFilter) {
+				files = files.filter(advancedFilter);
```
Comment (CarlSchwan) [inline core/src/OC/dialogs.js:1151]:
```suggestion
				files = files.filter(advancedFilter)
```

**Review by CarlSchwan — APPROVED** [review]

**Comment by PVince81** [comment]
I just did the bot's job

**Review by Pytal — APPROVED** [review]

**Comment by PVince81** [comment]
failing tests unrelated (the test server could not be started for 2 sharing related cases)

**Pull request merged**